src/sortable.js

```diff
@@ -23,116 +23,110 @@ angular.module('ui.sortable', [])
                 opts[key] = null;
               }
-            }
-            // for this directive to work we have to attach some callbacks
-            angular.forEach(callbacks, addDummyOptionKey);
-
-            // only initialize it in case we have to
-            // update some options of the sortable
-            var optsDiff = null;
-
-            if (oldVal) {
-              // reset deleted options to default
-              var defaultOptions;
-              angular.forEach(oldVal, function(oldValue, key) {
-                if (!newVal || !(key in newVal)) {
-                  if (key in directiveOpts) {
-                    if (key === 'ui-floating') {
-                      opts[key] = 'auto';
-                    } else {
-                      opts[key] = patchSortableOption(key, undefined);
-                    }
-                    return;
-                  }
-                  
-                  if (!defaultOptions) {
-                    defaultOptions = angular.element.ui.sortable().options;
-                  }
-                  var defaultValue = defaultOptions[key];
-                  defaultValue = patchSortableOption(key, defaultValue);
+            });
 
-                  if (!optsDiff) {
-                    optsDiff = {};
-                  }
-                  optsDiff[key] = defaultValue;
-                  opts[key] = defaultValue;
+            // initialize it in case we have to update some options of the sortable
+            var optsDiff = {};
+
+            // reset deleted options to default
+            var defaultOptions = angular.element.ui.sortable().options;
+            angular.forEach(oldVal, function(oldValue, key) {
+              if (!newVal || !newVal[key]) {
+                if (directiveOpts[key]) {
+                  opts[key] = key === 'ui-floating' ? 'auto' : patchSortableOption(key);
+                  return;
                 }
-              });
-            }
+                opts[key] = optsDiff[key] = patchSortableOption(key, defaultOptions[key]);
+              }
+            });
+            
 
             // update changed options
             angular.forEach(newVal, function(value, key) {
               // if it's a custom option of the directive,
               // handle it approprietly
-              if (key in directiveOpts) {
+              if (directiveOpts[key]) {
                 if (key === 'ui-floating' && (value === false || value === true) && sortableWidgetInstance) {
                   sortableWidgetInstance.floating = value;
                 }
-
                 opts[key] = patchSortableOption(key, value);
                 return;
               }
-
-              value = patchSortableOption(key, value);
-
-              if (!optsDiff) {
-                optsDiff = {};
-              }
-              optsDiff[key] = value;
-              opts[key] = value;
+              opts[key] = optsDiff[key] = patchSortableOption(key, value);
```

Comment:
having a separate `optsDiff = optsDiff || {};` line everywhere `optsDiff` is used, would be more appropriate